compiler/src/yul/constructor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::yul::base;
+use yultsur::yul;
+
+/// Builds a code block that returns the runtime object.
+///
+/// TODO: Add real constructor code here.
+pub fn code() -> yul::Code {
+    yul::Code {
+        block: yul::Block {
+            statements: vec![
+                yul::Statement::VariableDeclaration(yul::VariableDeclaration {
+                    identifiers: vec![base::untyped_identifier("size")],
+                    expression: Some(yul::Expression::FunctionCall(yul::FunctionCall {
+                        identifier: base::untyped_identifier("datasize"),
+                        arguments: vec![base::untyped_identifier_expr(r#""runtime""#)],
+                    })),
+                }),
+                yul::Statement::Expression(yul::Expression::FunctionCall(yul::FunctionCall {
+                    identifier: base::untyped_identifier("datacopy"),
+                    arguments: vec![
+                        base::untyped_literal_expr("0"),
+                        yul::Expression::FunctionCall(yul::FunctionCall {
+                            identifier: base::untyped_identifier("dataoffset"),
+                            arguments: vec![base::untyped_literal_expr(r#""runtime""#)],
+                        }),
+                        base::untyped_identifier_expr("size"),
+                    ],
+                })),
+                yul::Statement::Expression(yul::Expression::FunctionCall(yul::FunctionCall {
+                    identifier: base::untyped_identifier("return"),
+                    arguments: vec![
+                        base::untyped_literal_expr("0"),
+                        base::untyped_identifier_expr("size"),
+                    ],
+                })),
+            ],
+        },
+    }
+}
```

Comment:
i'll assume the specifics here are yul conventions for the constructor

Reply:
Basically [this](https://solidity.readthedocs.io/en/v0.6.7/yul.html#specification-of-yul-object)

Reply:
thanks for the link -- i did end up there last night and saw where you got it from -- nice work on the constructor basics here!